src/tasks/task_types.hpp

```diff
@@ -74,6 +86,7 @@ class Task {
   std::function<TaskStatus()> func_;
   int calls_ = 0;
   const int interval_;
+  std::chrono::high_resolution_clock::time_point start_time_;
```

Comment:
This is not for profiling. It's for detecting if a task gets stuck. When I was developing and debugging this PR, I got into situations where a sending block didn't flag the boundary status of the receiver as arrived (or worse, where some MPI messages seemed to get lost). This resulted in the `ReceiveBoundaryBuffers` task to get stuck in an infinite loop. By noting the start time of a task, the infrastructure can now check for how long the task has been running, and abort if the runtime indicates that it got stuck in an infinite loop.

This is a very crude and somewhat dirty way of implementing a timeout detection mechanism, but it was very helpful for my debugging and so I thought I'd leave it since it might come in handy again.

I'll add a comment here to explain that it's for timeout detection.